oop3-abstract-morph.js

```diff
@@ -39,21 +114,21 @@ class Person {
         console.log(this.name + " is eating");
     }
 
-    sleep = () => {
+    sleep() {
         console.log(this.name + " is sleeping");
     }
 
-    code = function() {
+    code() {
         console.log(this.name + " is coding");
     }
 
-    repeat = function() {
+    repeat() {
         console.log(this.name + " is repeating the above steps, yet another time");
     }
 
     explain() {
         //this function should contain a console.log() explaining what you had to do to get the correct functions to work, and the reasoning behind what you did.
-        console.log("this explain method should contain explain what you had to do to get the correct functions to work, and the reasoning behind what you did.");
+        console.log("I converted everything to an arrow function in the below classes so that the lower class objects function calls would take precedence over everything. ");
```

Comment:
this is one way to do it but only works with ES6 JS. A better way to do it would be to make sure the child classes have methods with the same name and signature. That way they'd Override the ones from the parent.